src/EnergyPlus/DataSurfaces.cc

```diff
@@ -597,6 +597,62 @@ Real64 SurfaceData::get_average_height(EnergyPlusData &state) const
     return std::abs(averageHeight) / SinTilt;
 }
 
+void SurfaceData::make_hash_key(EnergyPlusData &state, const int SurfNum)
+{
+    calcHashKey = SurfaceCalcHashKey();
+    calcHashKey.Construction = Construction;
+    calcHashKey.Azimuth = round(Azimuth * 10.0) / 10.0;
+    calcHashKey.Tilt = round(Tilt * 10.0) / 10.0;
+    calcHashKey.Height = round(Height * 10.0) / 10.0;
+    calcHashKey.Zone = Zone;
+    calcHashKey.EnclIndex = SolarEnclIndex;
+    calcHashKey.TAirRef = state.dataSurface->SurfTAirRef(SurfNum);
+
+    auto extBoundCond = state.dataSurface->Surface(SurfNum).ExtBoundCond;
+    if (extBoundCond > 0) {
+        calcHashKey.ExtZone = state.dataSurface->Surface(extBoundCond).Zone;
+        calcHashKey.ExtEnclIndex = state.dataSurface->Surface(extBoundCond).SolarEnclIndex;
+    } else {
+        calcHashKey.ExtZone = 0;
+        calcHashKey.ExtEnclIndex = 0;
+    }
+
+    calcHashKey.ExtSolar = ExtSolar;
+    calcHashKey.ExtWind = ExtWind;
+    calcHashKey.ViewFactorGround = round(ViewFactorGround * 10.0) / 10.0;
+    calcHashKey.ViewFactorSky = round(ViewFactorSky * 10.0) / 10.0;
+
+    calcHashKey.HeatTransferAlgorithm = HeatTransferAlgorithm;
+    calcHashKey.IntConvCoeff = state.dataSurface->SurfIntConvCoeffIndex(SurfNum);
+    calcHashKey.ExtConvCoeff = state.dataSurface->SurfExtConvCoeffIndex(SurfNum);
+    calcHashKey.OSCPtr = OSCPtr;
+    calcHashKey.OSCMPtr = OSCMPtr;
+
+    calcHashKey.FrameDivider = FrameDivider;
+    calcHashKey.SurfWinStormWinConstr = state.dataSurface->SurfWinStormWinConstr(SurfNum);
+
+    calcHashKey.MaterialMovInsulExt = state.dataSurface->SurfMaterialMovInsulExt(SurfNum);
+    calcHashKey.MaterialMovInsulInt = state.dataSurface->SurfMaterialMovInsulInt(SurfNum);
+    calcHashKey.SchedMovInsulExt = state.dataSurface->SurfSchedMovInsulExt(SurfNum);
+    calcHashKey.SchedMovInsulInt = state.dataSurface->SurfSchedMovInsulInt(SurfNum);
+    calcHashKey.ExternalShadingSchInd = state.dataSurface->SurfExternalShadingSchInd(SurfNum);
+    calcHashKey.SurroundingSurfacesNum = state.dataSurface->SurfSurroundingSurfacesNum(SurfNum);
+    calcHashKey.LinkedOutAirNode = state.dataSurface->SurfLinkedOutAirNode(SurfNum);
+    calcHashKey.OutsideHeatSourceTermSchedule = OutsideHeatSourceTermSchedule;
+    calcHashKey.InsideHeatSourceTermSchedule = InsideHeatSourceTermSchedule;
+}
+
+void SurfaceData::set_representative_surface(EnergyPlusData &state, const int SurfNum)
+{
+    // Make hash key for this surface (used to determine uniqueness)
+    state.dataSurface->Surface(SurfNum).make_hash_key(state, SurfNum);
+    // Insert surface key into map. If key already exists, it will not be added.
+    state.dataSurface->RepresentativeSurfaceMap.insert({state.dataSurface->Surface(SurfNum).calcHashKey, SurfNum});
```

Comment:
Relatively minor in the scheme of the PR, but `insert` returns a "pair object whose first element is an iterator pointing either to the newly inserted element in the container or to the element whose key is equivalent". This means you can just use that returned iterator below `state.dataSurface->Surface(SurfNum).RepresentativeCalcSurfNum = returned_pair.first;`, instead of performing a `find` again.